crates/melody_compiler/src/lib.rs

```diff
@@ -152,10 +154,18 @@ pub fn compiler(source: &str) -> Result<String, ParseError> {
 
             // modifiers
             Token::QuantifierExpression(quantity) => {
+                if in_modifier > 0 {
```

Comment:
```suggestion
                if in_modifier > 0 {
```

@Tigermouthbear I might be mistaken, but wouldn't `if quantifier.is_some()` work here rather than keeping new state?

Reply:
Yes, it would! I pushed a commit to this PR that makes these changes.